packages/headless/src/features/product-listing/product-listing-actions.ts

```diff
@@ -0,0 +1,144 @@
+import {createAsyncThunk, AsyncThunkAction} from '@reduxjs/toolkit';
+import {isErrorResponse} from '../../api/search/search-api-client';
+import {
+  CategoryFacetSection,
+  ConfigurationSection,
+  DateFacetSection,
+  FacetOptionsSection,
+  FacetOrderSection,
+  FacetSection,
+  NumericFacetSection,
+  PaginationSection,
+  ProductListingSection,
+  SortSection,
+} from '../../state/state-sections';
+import {getVisitorID} from '../../api/analytics/analytics';
+import {AnyFacetRequest} from '../facets/generic/interfaces/generic-facet-request';
+import {CategoryFacetSetState} from '../facets/category-facet-set/category-facet-set-state';
+import {sortFacets} from '../../utils/facet-utils';
+import {
+  AsyncThunkProductListingOptions,
+  ProductListingAPIClient,
+} from '../../api/commerce/product-listings/product-listing-api-client';
+import {
+  ProductListingRequest,
+  ProductListingSuccessResponse,
+} from '../../api/commerce/product-listings/product-listing-request';
+
+export type StateNeededByExecuteProductListing = ConfigurationSection &
+  ProductListingSection &
+  Partial<
+    PaginationSection &
+      SortSection &
+      FacetSection &
+      NumericFacetSection &
+      CategoryFacetSection &
+      DateFacetSection &
+      FacetOptionsSection &
+      FacetOrderSection
+  >;
+
+export interface ExecuteProductListingThunkReturn {
+  /** The successful search response. */
+  response: ProductListingSuccessResponse;
+  /** The number of milliseconds it took to receive the response. */
+  duration: number;
+}
+
+const fetchFromAPI = async (
+  client: ProductListingAPIClient,
+  _state: StateNeededByExecuteProductListing,
+  {request}: {request: ProductListingRequest}
+) => {
+  const startedAt = new Date().getTime();
+  const response = await client.getProducts(request);
+  const duration = new Date().getTime() - startedAt;
+  return {response, duration, requestExecuted: request};
```

Comment:
Is the duration important for product listing ?

We use it in the classic search use case for analytics logging, but we actually removed it from the UI (we don't display request time anymore, after a discussion with UX).

So, unless it's absolutely needed, I would suggest dropping all of this, since you were saying that no UA tracking should be required.

And if that assumption changes in the future, it will be trivial to add it back.